### PR TITLE
Pull Request of Long Short-Term Memory (LSTM) from Scratch

### PR DESCRIPTION
**Pull request by keshripritesh**
### 🔹 Summary

This PR adds an implementation of a **Long Short-Term Memory (LSTM)** model built completely **from scratch** without using deep learning frameworks such as TensorFlow or PyTorch.
The project demonstrates how **short-term and long-term memory components** can be modeled mathematically to learn and process sequential data efficiently.

### 🔹 Key Features

* ✅ LSTM cell implemented from scratch with mathematical operations
* ✅ Training on **US Baby Names Dataset** for sequence learning
* ✅ Achieves **84% accuracy** on test evaluation
* ✅ Fully documented Jupyter Notebook implementation
* ✅ Clear reference to theoretical foundations

### 🔹 Dataset

* **US Baby Names Dataset**
  🔗 [[Download Dataset](https://www.kaggle.com/datasets/kaggle/us-baby-names?select=NationalNames.csv)](https://www.kaggle.com/datasets/kaggle/us-baby-names?select=NationalNames.csv)

### 🔹 Tech Stack

* **Languages/Tools**: Python, Jupyter Notebook
* **Libraries**: Pandas, Numpy, Matplotlib

#### Prerequisites

Install the required dependencies:

```bash
pip install pandas
pip install numpy
pip install matplotlib
```

#### Running the Project

1. Clone the repository

   ```bash
   git clone <repo-url>
   cd <repo-folder>
   ```
2. Open the notebook

   ```bash
   jupyter notebook lstm_from_scratch.ipynb
   ```
3. Run all cells to train and evaluate the model

### 🔹 Model Performance

* The model achieves an **accuracy of 84%**, showcasing the effectiveness of the from-scratch LSTM implementation.

### 🔹 References

* Implementation ideas and explanations inspired by this [[Medium article](https://medium.com/hackernoon/understanding-architecture-of-lstm-cell-from-scratch-with-code-8da40f0b71f4)](https://medium.com/hackernoon/understanding-architecture-of-lstm-cell-from-scratch-with-code-8da40f0b71f4).

✨ This PR contributes a fully **from-scratch implementation of LSTM** with dataset preprocessing, training, and evaluation.

---

### CODE REVIEW

**Comment by MahraibFatima** [comment]
org GSSoC or SSoC?
